test/lib/next-test-utils.ts

```diff
@@ -543,6 +543,23 @@ export async function killApp(
   }
 }
 
+async function startListen(server: http.Server, port?: number) {
+  const listenerPromise = new Promise((resolve) => {
+    server['__socketSet'] = new Set()
```

Comment:
Maybe use a `WeapMap` instead?

Reply:
I expect server instance itself should goes away. If we have a wekmap that drops socket but server still stays that's problem need to be fixed.